### PR TITLE
Refactor to store multiple scrects, use uuid, tags support

### PR DESCRIPTION
**Pull request by sepulworld**
@sirech thanks for the ideas and help here! See: https://github.com/spulec/moto/issues/1893 

* Refactor for multiple secrets @sirech
* Use UUID replace hardcoded value
* Use random string in ARN id value
* Fix datetime stamp
* Adding support for Tags https://github.com/spulec/moto/issues/2004
* Add tests

### CODE REVIEW

File: moto/secretsmanager/models.py [hunk 1]
```diff
@@ -29,51 +30,57 @@ class SecretsManagerBackend(BaseBackend):
...
+        self.secrets = {}
 
     def reset(self):
         region_name = self.region
         self.__dict__ = {}
         self.__init__(region_name)
 
     def _is_valid_identifier(self, identifier):
-        return identifier in (self.name, self.secret_id)
+        if identifier in self.secrets:
```
Comment (sirech) [inline moto/secretsmanager/models.py]: would `return identifier in self.secrets` work?

File: moto/secretsmanager/models.py [hunk 2]
```diff
@@ -29,51 +30,57 @@ class SecretsManagerBackend(BaseBackend):
...
         })
 
         return response
 
     def create_secret(self, name, secret_string, **kwargs):
 
-        self.secret_string = secret_string
-        self.secret_id = name
-        self.name = name
+        self.generated_version_id = str(uuid.uuid4())
```
Comment (sirech) [inline moto/secretsmanager/models.py]: why is this stored as an instance variable? Shouldn't it be just a regular variable?

**Comment by sirech** [comment]
looks good to me!

File: moto/secretsmanager/models.py (hunk 2 above)
Comment (sepulworld) [reply moto/secretsmanager/models.py]: Good catch! Switching to a regular variable

File: moto/secretsmanager/models.py (hunk 1 above)
Comment (sepulworld) [reply moto/secretsmanager/models.py]: Yes, much cleaner! ty

**Comment by sepulworld** [comment]
@spulec this is good to go! Looks like the travis test failed due to an intermittent test failure with x-ray test suite.

**Comment by damacus** [comment]
Confirmed this also fixes the secrets http endpoint not returning the correct datetime.

**Comment by sirech** [comment]
I was wondering, is this one ready to be merged? I would really like to use it :)

**Comment by spulec** [comment]
Thanks for the help!

**Comment by TopherGopher** [comment]
@spulec I saw this was merged in a bit ago (awesome). What is your timeline for releases?

**Comment by djmrr** [comment]
@TopherGopher this is rad, also curious about a release timeline :)

**Pull request merged**